### ISSUE TITLE
Separate docs derivation to reduce build closure size

### ISSUE THREAD

**Issue by roberth** [NixOS/nix#7420]
**Is your feature request related to a problem? Please describe.**

A minimal NixOS configuration with `documentation.enable = false;` has a build dependency on mdbook through nix, because Nix is packaged as a single derivation.

**Describe the solution you'd like**

Package the docs in a separate derivation, and present the docs derivation like an output in the `nix` package attrset.
Blocked by #6507 

**Describe alternatives you've considered**

Either
 - Allow the docs to be disabled. This will cause multiple versions of Nix to be used.
 - Special-case the Nix package in NixOS to handle this.

Both alternatives burden the user with complexity.

**Additional context**

This is about the _derivation_ closure. The goal is to reduce dependencies and reduce rebuilds, not necessarily to reduce the closure of the build _outputs_.

**Priorities**

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by edolstra** [comment]
One thing to consider is that we may still want to start using Rust inside Nix in the future, in which case the mdbook dependency wouldn't be a big issue anymore.

**Comment by roberth** [comment]
Yeah, this is just a minor improvement that would be enabled by #6507. Other packages may benefit more from that change.

**Comment by Ericson2314** [comment]
Even if Nix use Rust, I think it is good to have finer-grained deps/derivations on principle.
